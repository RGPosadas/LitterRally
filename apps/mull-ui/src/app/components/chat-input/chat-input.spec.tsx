import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import ChatInput from './chat-input';

describe('ChatInput', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<ChatInput />);

    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<ChatInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
