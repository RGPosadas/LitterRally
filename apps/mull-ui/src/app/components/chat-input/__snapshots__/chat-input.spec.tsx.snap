// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput should match snapshot 1`] = `
<div
  className="chat-input-container"
  id="chat-input-container"
>
  <form
    autoComplete="off"
    className="chat-input-form"
    data-testid="chat-input"
    id="chat-input-form"
    onSubmit={[MockFunction]}
  >
    <div
      className="custom-file-upload-container"
    >
      <label
        className=""
        htmlFor="imageFile"
      >
        <div
          className="button file-upload-feedback"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-images fa-w-18 custom-file-upload-icon"
            data-icon="images"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M480 416v16c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V176c0-26.51 21.49-48 48-48h16v208c0 44.112 35.888 80 80 80h336zm96-80V80c0-26.51-21.49-48-48-48H144c-26.51 0-48 21.49-48 48v256c0 26.51 21.49 48 48 48h384c26.51 0 48-21.49 48-48zM256 128c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-96 144l55.515-55.515c4.686-4.686 12.284-4.686 16.971 0L272 256l135.515-135.515c4.686-4.686 12.284-4.686 16.971 0L512 208v112H160v-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </label>
      <input
        accept="image/*"
        className="custom-file-upload-input"
        data-testid="file"
        id="imageFile"
        type="file"
      />
    </div>
    <div
      className="mull-text-area-container "
    >
      <label
        className="mull-text-area-label"
        htmlFor="chat-input-textarea"
      >
        
      </label>
      <div
        className="mull-text-area-sub-container grow-wrap"
      >
        <div
          className="mull-text-area input-border "
          contentEditable={true}
          data-testid="mull-text-area"
          id="chat-input-textarea"
          onInput={[Function]}
          role="textbox"
          suppressContentEditableWarning={true}
        />
        <div
          className="mull-text-area-icon"
        >
          <button
            className="chat-input-button"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-paper-plane fa-w-16 "
              data-icon="paper-plane"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M476 3.2L12.5 270.6c-18.1 10.4-15.8 35.6 2.2 43.2L121 358.4l287.3-253.2c5.5-4.9 13.3 2.6 8.6 8.3L176 407v80.5c0 23.6 28.5 32.9 42.5 15.8L282 426l124.6 52.2c14.2 6 30.4-2.9 33-18.2l72-432C515 7.8 493.3-6.8 476 3.2z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
